frontend/app/utils/string-utils.ts

```diff
@@ -37,3 +39,97 @@ export function padWithZero(value: number, maxLength: number): string {
 export function trimToUndefined(str: string | undefined): string | undefined {
   return Number(str?.trim().length) > 0 ? str : undefined;
 }
+
+/**
+ * Arguments for formatting just the address line
+ */
+interface FormatAddressLineArguments {
+  /** Street address */
+  address: string;
+  /** Apartment, suite, or unit number */
+  apartment?: string;
+}
+
+/**
+ * Formats an apartment/suite number with the address
+ * @param params Address line formatting parameters
+ * @returns Formatted address line
+ */
+export function formatAddressLine({ address, apartment }: FormatAddressLineArguments): string {
+  if (!apartment?.trim()) {
+    return address;
+  }
+
+  // Check if apartment is a simple alphanumeric suite number
+  const isSuiteNumber = /^[a-z\d]+$/i.test(apartment.trim());
+
+  return isSuiteNumber ? `${apartment.trim()}-${address}` : `${address} ${apartment.trim()}`;
+}
+
+/**
+ * Arguments for formatting a complete address
+ */
+export interface FormatAddressArguments {
+  /** Street address */
+  address?: string;
```

Comment:
If this represents the street line of the address, then I think it should be named `street`. In my opinion, "address" represents the entire dataset being passed in, which includes not just the street name, but also the city, country, etc.